source/agora/network/Manager.d

```diff
@@ -760,6 +761,17 @@ public class NetworkManager
             return true;
         }
 
+        try
+        {
+            auto registry = (cast(FullNode) this.owner_node).getRegistry();
+            if (registry !is null)
+                foreach(addr; registry.getValidatorsAddresses())
+                    this.addAddress(addr);
```

Comment:
It's a bit bothering that we would:
- `cast` an interface to a `class` type (that's a failure of encapsulation);
- Expose an internal member of a class via `public` (I already wasn't happy with `package`);

Reply:
You could easily pass these to `discover()`, as parameters (which is already called by the `owner_node`). No need to cast or expose internals.